modules/python/examples/ukf-nonlinear-complex-example.py

```diff
@@ -0,0 +1,536 @@
+  @param b The other state vector that must be substracted to a.
+
+  @return ColVector The substraction a - b.
+  """
+  res = a - b
+  return ColVector([res[0], res[1], normalizeAngle(res[2])])
+
+def fx(x: ColVector, dt: float) -> ColVector:
+  """
+  @brief As the state model {x, y, \f$ \theta \f$} does not contain any velocity
+  information, it does not evolve without commands.
+
+  @param x The internal state of the UKF.
+  @param dt The sampling time: how far in the future are we projecting x.
+
+  @return ColVector The updated internal state, projected in time, also known as the prior.
+  """
+  return x
+
+def generateTurnCommands(v: float, angleStart: float, angleStop: float, nbSteps: int) -> List[ColVector]:
+  """
+  @brief Compute the commands realinp.sing a turn at constant linear velocity.
+
+  @param v Constant linear velocity.
+  @param angleStart Starting angle (in degrees).
+  @param angleStop Stop angle (in degrees).
+  @param nbSteps Number of steps to perform the turn.
+  @return List[ColVector] The corresponding list of commands.
+  """
+  cmds = []
+  dTheta = Math.rad(angleStop - angleStart) / float(nbSteps - 1);
+  for i in range(nbSteps):
+    theta = Math.rad(angleStart) + dTheta * float(i);
+    cmd = ColVector([v, theta])
+    cmds.append(cmd)
+  return cmds
+
+def generateCommands() -> List[ColVector]:
+  """
+  @brief Generate the list of commands for the simulation.
+
+  @return List[ColVector] The list of commands to use in the simulation
+  """
+  cmds = []
+  # Starting by an straight line acceleration
+  nbSteps = 30
+  dv = (1.1 - 0.001) / float(nbSteps - 1)
+  for i in range(nbSteps):
+    cmd = ColVector([0.001 + float(i) * dv, 0.])
+    cmds.append(cmd)
+
+  # Left turn
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  leftTurnCmds = generateTurnCommands(lastLinearVelocity, 0, 2, 15)
+  cmds.extend(leftTurnCmds)
+  for i in range(100):
+    cmds.append(cmds[len(cmds) -1])
+
+  # Right turn
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  rightTurnCmds = generateTurnCommands(lastLinearVelocity, 2, -2, 15);
+  cmds.extend(rightTurnCmds)
+  for i in range(200):
+    cmds.append(cmds[len(cmds) -1])
+
+  # Left  turn again
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  leftTurnCmds = generateTurnCommands(lastLinearVelocity, -2, 0, 15)
+  cmds.extend(leftTurnCmds)
+  for i in range(150):
+    cmds.append(cmds[len(cmds) -1])
+
+  lastLinearVelocity = cmds[len(cmds) -1][0]
+  leftTurnCmds = generateTurnCommands(lastLinearVelocity, 0, 1, 25)
+  cmds.extend(leftTurnCmds)
+  for i in range(150):
+    cmds.append(cmds[len(cmds) -1])
+
+  return cmds
+
+class vpBicycleModel:
+  """
+  @brief Class that approximates a 4-wheel robot unp.sing a bicycle model.
+  """
+  def __init__(self, w: float):
+    """
+    @brief Construct a new vpBicycleModel object.
+
+    @param w The length of the wheelbase.
+    """
+    self.m_w = w # The length of the wheelbase.
```

Comment:
In python, the convention for attributes that should not be visible is to prefix them with _ instead of m_